Fails for `g++`:

```
dump.c:1875:13: error: designator order for field ‘Perl_OpDumpContext::bar’ does not match declaration order in ‘Perl_OpDumpContext’
```

It seems in C++, the order of the named initialisers has to match the order of the fields in the definition. Which seems suuuuuper-annoying as surely the entire point of named initialisers is to remove that interdependence.. Grr. :(

--- a/dump.c
+++ b/dump.c
@@ -703,9 +703,9 @@ S_opdump_indent(pTHX_ const OP *o, I32 level, UV bar, PerlIO *file,
 }
 
 struct Perl_OpDumpContext {
-    I32 level;
-    UV bar;
     PerlIO *file;
+    UV bar;
+    I32 level;
     bool indent_needed;
 };
 
@@ -1868,9 +1868,9 @@ S_do_op_dump_bar(pTHX_ I32 level, UV bar, PerlIO *file, const OP *o,
 
         if(custom_dumper) {
             struct Perl_OpDumpContext ctx = {
-                .level         = level,
-                .bar           = bar,
                 .file          = file,
+                .bar           = bar,
+                .level         = level,
                 .indent_needed = true,
             };
             (*custom_dumper)(aTHX_ o, &ctx);

